File: code_samples/LinkedList/BasicLinkedList/BasicLinkedList.py
# Implémentation Python de la liste chaînée

class Node:
   #Création de noeud
    def __init__(self, item):
        self.item = item
        self.next = None


class LinkedList:

    def __init__(self):
        self.head = None


if __name__ == '__main__':

    linked_list = LinkedList()

    # Assignage de valeur
    linked_list.head = Node(1)
    second = Node(2)
    third = Node(3)

    # Connecter les noeuds
    linked_list.head.next = second
    second.next = third

    # Afficher la liste et ses items
    while linked_list.head != None:
        print(linked_list.head.item, end=" ")
        linked_list.head = linked_list.head.next

File: code_samples/LinkedList/BasicLinkedList/BasicLinkedList.cpp
// Implémentation C++ de la liste chaînée
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

// Création de noeud
class Node {
   public:
  int value;
  Node* next;
};

int main() {
  Node* head;
  Node* one = NULL;
  Node* two = NULL;
  Node* three = NULL;

  // allouer 3 noeuds
  one = new Node();
  two = new Node();
  three = new Node();

  // Assigner leurs valeurs
  one->value = 1;
  two->value = 2;
  three->value = 3;

  // Connecter les noeuds
  one->next = two;
  two->next = three;
  three->next = NULL;

  // Afficher la liste et ses valeurs
  head = one;
  while (head != NULL) {
    cout << head->value;
    head = head->next;
  }
}
